internal/configfsm/fsm.go

```diff
@@ -0,0 +1,309 @@
+package configfsm
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/qmuntal/stateless"
+)
+
+type Trigger string
+
+const (
+	// Transitions to Missing state
+	StartToMissing           Trigger = "Start->Missing"
+	ClusterNotReadyToMissing Trigger = "ClusterNotReady->Missing"
+	InProgressToMissing      Trigger = "InProgress->Missing"
+	CompletedToMissing       Trigger = "Completed->Missing"
+	TimedOutToMissing        Trigger = "TimedOut->Missing"
+	OutOfDateToMissing       Trigger = "OutOfDate->Missing"
+
+	// Transitions to ClusterNotReady state
+	MissingToClusterNotReady    Trigger = "Missing->ClusterNotReady"
+	InProgressToClusterNotReady Trigger = "InProgress->ClusterNotReady"
+	CompletedToClusterNotReady  Trigger = "Completed->ClusterNotReady"
+	OutOfDateToClusterNotReady  Trigger = "OutOfDate->ClusterNotReady"
```

Comment:
Could we add a small comment explaining why `TimedOut -> ClusterNotReady `is not a viable transition, please?